raiden/network/rpc/client.py

```diff
@@ -244,46 +229,25 @@ def block_number(self):
         """ Return the most recent block. """
         return self.web3.eth.blockNumber
 
-    def nonce_needs_update(self):
-        if self.nonce_available_value is None:
-            return True
-
-        now = time.time()
-
-        # Python's 2.7 time is not monotonic and it's affected by clock resets,
-        # force an update.
-        if self.nonce_last_update > now:
-            return True
-
-        return now - self.nonce_last_update > self.nonce_update_interval
-
-    def nonce_update_from_node(self):
-        nonce = -2
-        nonce_available_value = self.nonce_available_value or -1
-
-        # Wait until all tx are registered as pending
-        while nonce < nonce_available_value:
-            pending_transactions = self.web3.eth.getTransactionCount(
-                to_checksum_address(self.sender),
-            )
-            nonce = pending_transactions + self.nonce_offset
-
-            log.debug(
-                'updated nonce from server',
-                server=nonce,
-                local=nonce_available_value,
-            )
+    @cachedmethod(attrgetter('_nonce_cache'))
+    def _node_nonce_it(self) -> Iterable[int]:
```

Comment:
Just for the record, I think that having regular nonce updates is more of a problem then an advantage. It can't fix any nonce races across applications, and it has introduced quite a few bugs in the RPC Client in the past.

Reply:
Race across application you mean using the same ethereum account outside Raiden and thus having unpredictable nonces, right? We have it [documented](http://raiden-network.readthedocs.io/en/latest/overview_and_guide.html#firing-it-up) as a restriction which if not followed can break things.

But then .... I guess what you are trying to say is why query the ethereum node for the nonce at all? Since we have set the restriction we are the only ones who are going to be using the ethereum account then we can keep our own counter.

Is that what you mean?

Reply:
> Race across application you mean using the same ethereum account outside Raiden and thus having unpredictable nonces, right? We have it documented as a restriction which if not followed can break things

Yes. I believe this was documented exactly because of the problem we have had with nonces in the past.

> But then .... I guess what you are trying to say is why query the ethereum node for the nonce at all? Since we have set the restriction we are the only ones who are going to be using the ethereum account then we can keep our own counter.

Yes, it's logic that doesn't fix any problems, at best it hides problems.